### ISSUE TITLE
Using R-devel when the version bumps before Bioconductor's version bumps?

### ISSUE THREAD

**Issue by bbimber** [Bioconductor/BiocManager#130]
Hello,

We have a github actions workflow configured to use r-devel, and bioc-devel. If I'm interpreting things correctly, r-devel just switched to report version 4.3, and bioconductor will not install on r-devel since it supports 4.2. Is there a way to force bioconductor to install on the development version of R, even though it reports a version higher than bioconductor officially supports?

**Comment by bbimber** [comment]
I believe my question is similar to this issue, which is not surprisingly about one year ago:

https://github.com/Bioconductor/BiocManager/issues/99

**Comment by mtmorgan** [comment]
This situation will arise (correctly, in my opinion) every year. 'bioc-devel' anticipates the software that users will have when the 'bioc-devel' version roles over to the next release. Since Bioconductor has a 6 month release cycle, and R has an annual release cycle, there is a 6 month period (starting now) when R-devel is irrelevant to the anticipated Bioconductor release experience.

You could take the philosophical perspective that you're using base R, and not Bioconductor, and follow the base R procedures for installing Bioconductor repositories using, e.g.,
```
setRepositories(ind = 1:5)
```
and then
```
> options("repos")
$repos
                                                    CRAN
                                                "@CRAN@"
                                                BioCsoft
           "https://bioconductor.org/packages/3.15/bioc"
                                                 BioCann
"https://bioconductor.org/packages/3.15/data/annotation"
                                                 BioCexp
"https://bioconductor.org/packages/3.15/data/experiment"
                                               CRANextra
                   "https://www.stats.ox.ac.uk/pub/RWin"
```
Then use `install.packages()` to install CRAN and Bioconductor packages. `BiocManager::valid()` will return `FALSE`.

I think it's appropriate for `BiocManager::install()` to fail on the the current R-devel -- sure we could install packages, but they wouldn't have been built (especially important for binary macOS and Windows packages) and checked on R-devel.

**Comment by bbimber** [comment]
@mtmorgan I can see that to a point. Separately, I'm exploring whether rlib-actions/setup can be pointed at the 4.2 pre-release (which I agree with you is the correct target for Bioc-devel). If I'm interpreting right, the issue is not so much the fact that bioc-devel and r-devel dont match up anymore, but rather getting a github actions to point at an R release compatible with bioc-devel. This is not a BioManager issue per se.

**Comment by hpages** [comment]
The current BioC devel is 3.15 and it's tied to R 4.2. R-devel (which always points to trunk in svn) is no longer R 4.2: earlier this week it became R 4.3. At the same time the R core team created the R 4.2 branch in svn. The daily snapshots of the R 4.2 branch are made available [here](https://cran.r-project.org/src/base-prerelease/) (`R-alpha-*.tar.gz` tarballs).
 
 I don't know how you specify what version of R to use in GitHub actions, but for BioC 3.15 packages (current devel), you want to point to the latest R-alpha/R-beta/R-RC snapshot available at this location. Not easy to do programmatically given that the exact name of the tarball is hard to predict and CRAN doesn't seem to provide a fixed URL for this (via a symlink or something like that).

H.

**Comment by bbimber** [comment]
One final thought here: there is a clear argument that rlib/actions should have some way to install R 4.2 alpha/beta/RC (as opposed to just 'devel'). That would mitigate these interim periods.

However, the fact that Bioconductor is failing is somewhat artificial here. I agree with @mtmorgan that the right thing to do in a default case is to fail, like it does. However, it seems like it would also be appropriate to support some kind of argument/environment variable that says 'yes, i understand the R version doesnt match, install anyway'. For the purposes of things like automated testing, it seems pretty reasonable, and would save some headache, for the developer to force Bioconductor to run.

**Comment by LiNk-NY** [comment]
In the `DESCRIPTION` file of `BiocVersion`, we say `Depends: R (>= 4.2.0)` but in `BiocManager` code we have limits (i.e., && `R (< 4.3.0)`) as to how far that `>=` extends. There may be changes that are implemented immediately in `4.3.0` that would affect the code behavior. Our 'guarantee' is defined within that interval and maybe it should be made more explicit (though the error in `BiocManager` I think is appropriate and sufficient).

In a way, adding an environment variable override would be against what we believe to be true and tested in Bioc-devel.
We've always had a lag between the available containers and releases of R for Docker versions of Bioconductor and Continuous Integration systems (e.g., GitHub Actions, AppVeyor, etc.). I think the strategy has been to wait for those containers to get built before testing.
